app/uk/gov/hmrc/gform/gform/FormComponentUpdater.scala

```diff
@@ -75,7 +75,11 @@ class FormComponentUpdater(formComponent: FormComponent, index: Int, baseIds: Li
     shortName = formComponent.shortName.map(expandSmartString),
     errorMessage = formComponent.errorMessage.map(expandSmartString),
     validators = formComponent.validators.map(expandFormComponentValidator),
-    instruction = formComponent.instruction.map(i => i.copy(name = expandSmartString(i.name)))
+    instruction = formComponent.instruction.map(i =>
+      i.name match {
+        case Some(name) => i.copy(name = Some(expandSmartString(name)))
+        case _          => i
+    })
```

Comment:
What about to do rather:

```suggestion
    instruction = formComponent.instruction.map(i => i.copy(name = i.name.map(expandSmartString)))
```

Reply:
good suggestion, done